### PR TITLE
[blockindex] remove deprecated DeleteTipBlock

### PR DESCRIPTION
**Pull request by envestcc**
# Description

DeleteTipBlock has been deprecated, removing it from blockindexer

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by dustinxie** [comment]
should have more places to remove this func, you can refer to earlier #3854

File: blockchain/blockdao/blockindexer.go [hunk 1]
```diff
@@ -25,7 +25,6 @@ type (
 		Stop(ctx context.Context) error
 		Height() (uint64, error)
 		PutBlock(context.Context, *block.Block) error
-		DeleteTipBlock(context.Context, *block.Block) error
```
Comment (Liuhaai) [inline blockchain/blockdao/blockindexer.go:28]: https://github.com/iotexproject/iotex-core/pull/2647

**Review by Liuhaai — CHANGES_REQUESTED** [review]

File: blockchain/blockdao/blockindexer.go (hunk 1 above)
Comment (Liuhaai) [reply blockchain/blockdao/blockindexer.go:28]: lmk if the early issue has been resolved

**Comment by dustinxie** [comment]
we have 2 earlier PR (#2647 and #3854) trying to clean-up this.

IMO, `DeleteTipBlock` is a wrong method to start with, so it should be deleted. I propose let's use the earliest #2647 to finish this off. @Liuhaai  let us know

**Comment by envestcc** [comment]
> we have 2 earlier PR (#2647 and #3854) trying to clean-up this.
> 
> IMO, `DeleteTipBlock` is a wrong method to start with, so it should be deleted. I propose let's use the earliest #2647 to finish this off. @Liuhaai let us know

Okay, use the previous and close this one

**Pull request closed without merging**